RevenueCatUI/Views/RemoteImage.swift

```diff
@@ -20,51 +20,67 @@ struct RemoteImage: View {
     let aspectRatio: CGFloat?
     let maxWidth: CGFloat?
 
+    @StateObject
+    private var loader: ImageLoader = .init()
+
     init(url: URL, aspectRatio: CGFloat? = nil, maxWidth: CGFloat? = nil) {
         self.url = url
         self.aspectRatio = aspectRatio
         self.maxWidth = maxWidth
     }
 
     var body: some View {
-        AsyncImage(
-            url: self.url,
-            transaction: .init(animation: Constants.defaultAnimation)
-        ) { phase in
-            if let image = phase.image {
+        Group {
+            switch self.loader.result {
+            case .none:
+                self.emptyView(error: nil)
+
+            case let .success(image):
                 if let aspectRatio {
                     image
                         .fitToAspect(aspectRatio, contentMode: .fill)
                         .frame(maxWidth: self.maxWidth)
+
                 } else {
                     image
                         .resizable()
                 }
+
+            case let .failure(error):
+                self.emptyView(error: error)
+            }
+        }
+        .transition(Self.transition)
+        .task(id: self.url) {
```

Comment:
This automatically cancels the task when the url changes.

Reply:
🤔 let's maybe add a comment? Feels like that would be easy to miss

Reply:
It's the canonical way of implementing task: making the ID whatever parameter the task depends on. But yeah I'll add a comment 👍🏻